### PR TITLE
add basic support for overrideLabel in extract_subgraphs_from_supergraph

### PR DESCRIPTION
**Pull request by lennyburdette**
---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Comment by dariuszkuc** [comment]
Do we also need a guard that this is only applicable to fed 2.7+ version?

**Comment by lennyburdette** [comment]
i can add a guard but since this is just for the in-memory versions of subgraphs extracted from the supergraph, i don't think it matters?

File: apollo-federation/src/merge.rs [hunk 1]
```diff
@@ -510,6 +515,17 @@ impl Merger {
                     },
                 );
 
+                let overrides_directive_option = Option::and_then(
+                    field.directives.get_all(&override_directive_name).next(),
+                    |p| {
+                        let overrides_from =
+                            directive_string_arg_value(p, &name!("from")).map(|v| v.as_str());
+                        let ovevrrides_label =
+                            directive_string_arg_value(p, &name!("label")).map(|v| v.as_str());
```
Comment (dylan-apollo) [inline apollo-federation/src/merge.rs:524]: Should these use the fed spec consts?

File: apollo-federation/src/merge.rs [hunk 2]
```diff
@@ -510,6 +515,17 @@ impl Merger {
                     },
                 );
 
+                let overrides_directive_option = Option::and_then(
+                    field.directives.get_all(&override_directive_name).next(),
```
Comment (dylan-apollo) [inline apollo-federation/src/merge.rs:519]: Curious why this vs

`field.directives.get_all(&override_directive_name).next().and_then(...)1

File: apollo-federation/src/merge.rs [hunk 3]
```diff
@@ -1088,6 +1105,13 @@ fn join_field_directive_definition() -> DirectiveDefinition {
                 ty: ty!(String).into(),
                 default_value: None,
             }),
+            Node::new(InputValueDefinition {
+                name: name!("overrideLabel"),
```
Comment (dylan-apollo) [inline apollo-federation/src/merge.rs:1109]: Same Q re: const

**Review by dylan-apollo — APPROVED** [review]
I don't know anything about join spec, so only reviewed for style 😁

File: apollo-federation/src/merge.rs (hunk 2 above)
Comment (lennyburdette) [reply apollo-federation/src/merge.rs:519]: just how it was already written! i can definitely change it

File: apollo-federation/src/merge.rs (hunk 3 above)
Comment (lennyburdette) [reply apollo-federation/src/merge.rs:1109]: it's a fair question but this whole file is filled with strings and i don't really want to make a bigger diff than necessary! i'll replace this one at least

File: apollo-federation/src/merge.rs (hunk 2 above)
Comment (lennyburdette) [reply apollo-federation/src/merge.rs:519]: lol Option::and_then avoided a clippy warning

**Pull request merged**